python/mxnet/symbol/symbol.py

```diff
@@ -2627,6 +2633,24 @@ def detach(self):
     def backward(self):
         raise NotImplementedForSymbol(self.backward, None)
 
+    def optimize_for_dynamic_shape_op(self, is_np_sym=False):
```

Comment:
Why is this a public API even though it's called automatically in `_build_cache`? Should it be private?

Reply:
Will change to private function